app/api/threads/list_threads.feature

```diff
@@ -0,0 +1,172 @@
+    And AlbertEinstein the scientist is a member of the group Laboratory
+    And AlbertEinstein has approved access to his personal info for the group Laboratory
+    And PaulDirac the scientist is a member of the group Laboratory
+    And the database has the following table 'items':
+      | id | type | default_language_tag |
+      | 1  | Task | fr                   |
+      | 2  | Task | en                   |
+    And the database has the following table 'items_strings':
+      | item_id | language_tag | title      |
+      | 1       | en           | Beginning  |
+      | 1       | fr           | Debut      |
+      | 2       | en           | Experiment |
+    And the database has the following table 'threads':
+      | item_id | participant_id  | status                  | message_count | latest_update_at    | helper_group_id |
+      | 1       | @AlbertEinstein | waiting_for_trainer     | 0             | 2023-01-01 00:00:01 | @Laboratory     |
+      | 2       | @PaulDirac      | waiting_for_participant | 1             | 2023-01-01 00:00:02 | @Laboratory     |
+    When I send a GET request to "/threads?watched_group_id=@Laboratory"
+    Then the response code should be 200
+    And the response body should be, in JSON:
+    """
+      [
+        {
+          "item": {
+            "id": "1",
+            "language_tag": "fr",
+            "title": "Debut",
+            "type": "Task"
+          },
+          "latest_update_at": "2023-01-01T00:00:01Z",
+          "message_count": 0,
+          "participant": {
+            "id": "@AlbertEinstein",
+            "login": "AlbertEinstein",
+            "first_name": "Albert",
+            "last_name": "Einstein"
+          },
+          "status": "waiting_for_trainer"
+        },
+        {
+          "item": {
+            "id": "2",
+            "language_tag": "en",
+            "title": "Experiment",
+            "type": "Task"
+          },
+          "latest_update_at": "2023-01-01T00:00:02Z",
+          "message_count": 1,
+          "participant": {
+            "id": "@PaulDirac",
+            "login": "PaulDirac",
+            "first_name": "",
+            "last_name": ""
+          },
+          "status": "waiting_for_participant"
+        }
+      ]
+    """
+
+  Scenario: Should get the threads whose the participant is a descendant of the watched_group_id
+    Given I am RichardFeynman
+    And I can watch the group University
+    And the user Mary is referenced by @Mary
+    And the user PresidentUniversity is referenced by @PresidentUniversity
+    And the group University is referenced by @University
+    When I send a GET request to "/threads?watched_group_id=@University"
+    And it should be a JSON array with 2 entries
+    And the response should match the following JSONPath:
+      | JSONPath            | value                |
+      | $[*].participant.id | @PresidentUniversity |
+      | $[*].participant.id | @Mary                |
```

Comment:
Actually I used json path equivalent in gherkin tests for another project. That works well as long as you have a least an example of full response (as you did above).

I don't think there is a benefit from using a gherkin table anymore when using jsonpath (except if the value to check can be represented as a table with an actual benefit). I would be shorter/clearer to have rules like `And the response at $[*].participant.id should be "...." , and possibly have several of this rule.

EDIT: uhm here you want to check a list actually... so what what I would expect instead:
```
And the response at $[*].participant.id should be:
  | value                |
  | @PresidentUniversity |
  | @Mary                |
```

Reply:
fixed, there is a version for multiple checks with a table, and the inline for a single check